app/lib/features/tasks/sheets/create_update_task_list.dart

```diff
@@ -0,0 +1,199 @@
+      final spaceNotifier = ref.read(selectedSpaceIdProvider.notifier);
+      spaceNotifier.state = widget.initialSelectedSpace;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _buildBody(context);
+  }
+
+  Widget _buildBody(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 20.0),
+      child: SingleChildScrollView(
+        child: Form(
+          key: _formKey,
+          child: Column(
+            mainAxisSize: MainAxisSize.min,
+            crossAxisAlignment: CrossAxisAlignment.stretch,
+            children: [
+              const SizedBox(height: 20),
+              const Divider(indent: 150,endIndent: 150,thickness: 2,),
+              const SizedBox(height: 20),
+              Text(
+                L10n.of(context).createNewTaskList,
+                textAlign: TextAlign.center,
+                style: Theme.of(context).textTheme.titleMedium,
+              ),
+              const SizedBox(height: 20),
+              _widgetTaskListName(),
+              _widgetDescription(),
+              const SizedBox(height: 20),
+              const SelectSpaceFormField(canCheck: 'CanPostTaskList'),
+              _widgetMoreDetails(),
+              const SizedBox(height: 20),
+              _widgetCreateButton(),
+              const SizedBox(height: 20),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _widgetTaskListName() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          L10n.of(context).taskListName,
+          style: Theme.of(context).textTheme.bodySmall,
+        ),
+        const SizedBox(height: 5),
+        TextFormField(
+          autofocus: true,
+          key: CreateUpdateTaskList.titleKey,
+          decoration: InputDecoration(
+            hintText: L10n.of(context).name,
+          ),
+          autovalidateMode: AutovalidateMode.onUserInteraction,
+          controller: _titleController,
+          validator: (value) => (value?.isNotEmpty == true)
+              ? null
+              : L10n.of(context).pleaseEnterAName,
+        ),
+      ],
+    );
+  }
+
+  Widget _widgetDescription() {
+    return ValueListenableBuilder(
+      valueListenable: isShowDescription,
+      builder: (context, isShowDescriptionValue, child) {
+        if (!isShowDescriptionValue) return const SizedBox.shrink();
+        return Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            const SizedBox(height: 20),
+            Text(
+              L10n.of(context).description,
+              style: Theme.of(context).textTheme.bodySmall,
+            ),
+            MdEditorWithPreview(
+              key: CreateUpdateTaskList.descKey,
+              onChanged: (String? value) {
+                _descriptionText.value = value ?? '';
+              },
+            ),
```

Comment:
we are phasing out the MDEditorWithPreview in favor of the AppFlowyEditor. We should use that one here instead.

Reply:
Can we do this on separate PR as I first I would like to focus on Refactoring task UI-UX to quickly get out of lab feature and then improve individual smaller pieces on quick PRs.

Side-note:
- Not sure but probably also need to replace `taskListDraft.descriptionMarkdown` with `taskListDraft.descriptionHtml(plaintext, htmltext)`.